[fix] Fallback to system CA certs pool

yurishkuro opened the pull request:
## Which problem is this PR solving?
- Resolves #6334

## Description of the changes
- Restore pre-OTEL behavior of using system cert pool if no CAFile is specified.

## How was this change tested?
- unit tests

mahadzaryab1 reviewed (approved):
lgtm - there's a just failing unit test

yurishkuro commented:
the failing tests indicate larger problems with Kafka options handling, I see some bogus conditions being asserted, like `tls.enable=false` and yet expecting valid TLS config.

yurishkuro commented:
@mahadzaryab1 I fixed the tests. The business logic in the tests themselves is a bit funky and inconsistent, e.g. `auth=tls` and `tls.enabled=false` are not mutually exclusive, which seems weird. But I didn't want to invest more into this legacy part - as we switched to OTEL's components we can assume the validation to be happening here.
